verification/src/contextual_block_verifier.rs

```diff
@@ -248,22 +249,43 @@ impl<'a, 'b, CS: ChainStore<'a>> RewardVerifier<'a, 'b, CS> {
         }
     }
 
+    #[allow(clippy::int_plus_one)]
     pub fn verify(&self) -> Result<Vec<Capacity>, Error> {
         let cellbase = &self.resolved[0];
-        let (target_lock, block_reward) = self.context.finalize_block_reward(self.parent)?;
-        if cellbase.transaction.outputs_capacity()? != block_reward.total {
-            return Err((CellbaseError::InvalidRewardAmount).into());
+        let no_finalization_target =
+            (self.parent.number() + 1) <= self.context.consensus.finalization_delay_length();
+        if no_finalization_target {
+            let ret = if cellbase.transaction.outputs().is_empty() {
+                Ok(vec![])
+            } else {
+                Err((CellbaseError::InvalidRewardTarget).into())
+            };
+            return ret;
         }
-        if cellbase
-            .transaction
-            .outputs()
-            .get(0)
-            .expect("cellbase should have output")
-            .lock()
-            != target_lock
-        {
-            return Err((CellbaseError::InvalidRewardTarget).into());
+
+        let (target_lock, block_reward) = self.context.finalize_block_reward(self.parent)?;
+        let output = CellOutput::new_builder()
+            .capacity(block_reward.total.pack())
+            .lock(target_lock.clone())
+            .build();
+        let insufficient_reward_to_create_cell = output.is_lack_of_capacity(Capacity::zero())?;
+
+        if !insufficient_reward_to_create_cell {
+            if cellbase.transaction.outputs_capacity()? != block_reward.total {
+                return Err((CellbaseError::InvalidRewardAmount).into());
+            }
+            if cellbase
+                .transaction
+                .outputs()
+                .get(0)
+                .expect("cellbase should have output")
+                .lock()
+                != target_lock
+            {
+                return Err((CellbaseError::InvalidRewardTarget).into());
+            }
         }
```

Comment:
Add one more assertion that if it's `insufficient_reward_to_create_cell` the outputs must be empty.